### DISCUSSION TITLE
How to know the size of the thread pool ? And what is the use of --workers in uvicorn

### DISCUSSION THREAD

**Discussion by yyfhust** [fastapi/fastapi#4593] [Questions] [answered]
Hi,
How to know the size of the thread pool ? And what is the use of --workers in uvicorn

I have tested locally:
I used uvicorn to run fastAPI based backend locally on my Mac, and I can see that it can create up to 6 distinct threads ( the server gets blocked when I sent more request ).
I also tried to run the backend with different --workers, but there was no difference at all, the threads limit was always 6 no matter how many workers I specified.
Commend I have run:
uvicorn run:app --host 0.0.0.0 --port 8000 --workers 1
uvicorn run:app --host 0.0.0.0 --port 8000 --workers 2
uvicorn run:app --host 0.0.0.0 --port 8000 --workers 10

**Comment by Kludex** [comment, marked as answer]
Uvicorn doesn't manage threads.

On the other hand, this may be interesting to you: https://github.com/encode/starlette/issues/1724
